cmd/local-artifact-mirror/artifact.go

```diff
@@ -0,0 +1,80 @@
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"oras.land/oras-go/v2"
+	"oras.land/oras-go/v2/content/file"
+	"oras.land/oras-go/v2/registry"
+	"oras.land/oras-go/v2/registry/remote"
+	"oras.land/oras-go/v2/registry/remote/auth"
+)
+
+// registryAuth returns the credentials to reach the registry. These credentials are
+// read from the cluster. XXX It is still not defined the secret name from where we
+// are going to read the credentials so this returns empty credentials instead.
+func registryAuth(_ context.Context) (auth.Credential, error) {
```

Comment:
This is the piece we are missing here. Read the registry credentials from the cluster. We need to give enough permissions for the "embedded-cluster-operator" service account.